In [36]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import MinMaxScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix,classification_report
from sklearn.model_selection import train_test_split
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [37]:
data = pd.read_excel("C:/Users/Sadyo/Desktop/Final Participant Data Folder/Final_Train.xlsx")
data_test=pd.read_excel("C:/Users/Sadyo/Desktop/Final Participant Data Folder/Final_Test.xlsx")

In [38]:
data.head()

Qualification           Experience Rating  \
0           BHMS, MD - Homeopathy  24 years experience   100%   
1    BAMS, MD - Ayurveda Medicine  12 years experience    98%   
2  MBBS, MS - Otorhinolaryngology   9 years experience    NaN   
3             BSc - Zoology, BAMS  12 years experience    NaN   
4                            BAMS  20 years experience   100%   

                          Place         Profile  \
0           Kakkanad, Ernakulam       Homeopath   
1         Whitefield, Bangalore        Ayurveda   
2    Mathikere - BEL, Bangalore  ENT Specialist   
3  Bannerghatta Road, Bangalore        Ayurveda   
4         Keelkattalai, Chennai        Ayurveda   

                                  Miscellaneous_Info  Fees  
0               100% 16 Feedback Kakkanad, Ernakulam   100  
1              98% 76 Feedback Whitefield, Bangalore   350  
2                                                NaN   300  
3  Bannerghatta Road, Bangalore ₹250 Available on...   250  
4              100% 4 Feedback Keelkattalai, Chennai   250

It can be observed that to do mathematical work on this data it needs lot of Preprocessing.First I would try to extract meaningful information from data.

In [39]:
data.isnull().sum()

Qualification            0
Experience               0
Rating                3302
Place                   25
Profile                  0
Miscellaneous_Info    2620
Fees                     0
dtype: int64

In [40]:
data.shape

(5961, 7)

Lot of work has to be done on Rating, Place, Miscellaneous_Info columns. Try to not to drop any column as we already have less number of predictors.

In [41]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5961 entries, 0 to 5960
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Qualification       5961 non-null   object
 1   Experience          5961 non-null   object
 2   Rating              2659 non-null   object
 3   Place               5936 non-null   object
 4   Profile             5961 non-null   object
 5   Miscellaneous_Info  3341 non-null   object
 6   Fees                5961 non-null   int64 
dtypes: int64(1), object(6)
memory usage: 326.1+ KB


In [42]:
data_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1987 entries, 0 to 1986
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Qualification       1987 non-null   object
 1   Experience          1987 non-null   object
 2   Rating              897 non-null    object
 3   Place               1981 non-null   object
 4   Profile             1987 non-null   object
 5   Miscellaneous_Info  1153 non-null   object
dtypes: object(6)
memory usage: 93.3+ KB


In [43]:
# First, converting Rating into float or integer type :
  # following code would give the ratings in float data type:
    ## making calculations on that would be easy and effective as well:
    
data['Rating'] = data['Rating'].str.rstrip('%').astype('float') 
data_test['Rating'] = data_test['Rating'].str.rstrip('%').astype('float')

In [44]:
# converting the Experince column in integers:

data["Experience"] = data['Experience'].str.replace(r'\D', '')           # it will remove the text written with number of years
data_test["Experience"] = data_test['Experience'].str.replace(r'\D', '')

In [45]:
data["Experience"]

0       24
1       12
2        9
3       12
4       20
5        8
6       42
7       10
8       14
9       23
10       5
11       7
12       9
13      21
14      12
15      10
16      24
17       5
18      19
19      19
20      21
21      14
22      41
23      31
24      23
25       5
26      25
27      12
28      16
29      15
30       7
31      11
32      14
33      23
34       9
35      15
36      10
37      12
38      31
39      16
40       8
41      12
42      31
43       8
44      22
45       7
46      16
47      31
48       8
49      10
50      42
51       9
52      22
53      10
54       9
55      25
56      14
57      11
58      24
59      38
60       8
61      12
62      14
63       5
64      22
65      14
66       6
67      36
68       6
69      13
70      13
71      11
72      18
73      46
74      41
75      40
76       7
77      13
78      12
79      10
80      49
81       9
82      20
83      11
84      10
85      16
86      24
87      33
88      10
89      36
90       8

In [46]:
data_test["Experience"]

0       35
1       31
2       40
3        0
4       16
5       14
6       23
7        9
8       11
9       44
10       4
11      34
12      11
13      13
14       4
15       4
16      16
17      28
18       6
19      32
20      11
21       3
22       8
23      13
24      43
25      27
26      22
27       8
28      25
29       4
30       4
31      36
32      14
33      17
34      20
35       9
36       8
37      28
38      18
39      34
40       4
41      18
42      18
43      36
44      29
45      29
46       6
47       5
48      19
49       9
50      19
51      10
52      23
53      31
54       2
55      20
56       4
57      20
58      30
59      24
60      13
61      32
62      10
63      26
64      27
65      41
66      46
67      18
68      11
69      36
70      21
71      19
72      12
73       9
74       9
75      20
76       8
77      17
78      40
79      14
80      45
81      33
82      19
83       8
84       3
85      16
86      35
87      28
88      27
89      16
90      17

In [47]:
# checking the type of data of Experience:

data["Experience"].dtype

data_test["Experience"].dtype

dtype('O')

In [48]:
# converting datatype of Experience into integer:

data["Experience"] = data["Experience"].astype(str).astype(float) 
data_test["Experience"] = data_test["Experience"].astype(str).astype(float)

In [49]:
# as there is no Null values in Experience column, so it can be converted to integer wright now:

data["Experience"] = data["Experience"].astype(int)
data_test["Experience"] = data_test["Experience"].astype(int)

In [50]:
data_test["Experience"].dtype

dtype('int32')

In [51]:
data["Experience"].dtype

dtype('int32')

Now , treating Place column, it is difficult to deal with the present format of Place column : Trying splitting it into two columns :

In [52]:
# Following code will split the values of Place column into two seperate columns:

data['Place'], data['City'] = data['Place'].str.rsplit(', ', 1).str
data_test['Place'],data_test['City'] = data_test['Place'].str.rsplit(',',1).str

In [53]:
data.head()

Qualification  Experience  Rating              Place  \
0           BHMS, MD - Homeopathy          24   100.0           Kakkanad   
1    BAMS, MD - Ayurveda Medicine          12    98.0         Whitefield   
2  MBBS, MS - Otorhinolaryngology           9     NaN    Mathikere - BEL   
3             BSc - Zoology, BAMS          12     NaN  Bannerghatta Road   
4                            BAMS          20   100.0       Keelkattalai   

          Profile                                 Miscellaneous_Info  Fees  \
0       Homeopath               100% 16 Feedback Kakkanad, Ernakulam   100   
1        Ayurveda              98% 76 Feedback Whitefield, Bangalore   350   
2  ENT Specialist                                                NaN   300   
3        Ayurveda  Bannerghatta Road, Bangalore ₹250 Available on...   250   
4        Ayurveda              100% 4 Feedback Keelkattalai, Chennai   250   

        City  
0  Ernakulam  
1  Bangalore  
2  Bangalore  
3  Bangalore  
4    Chennai

In [54]:
data_test.head()

Qualification  Experience  Rating  \
0                                             MBBS          35     NaN   
1       MBBS, Diploma in Otorhinolaryngology (DLO)          31     NaN   
2                                       MBBS, DDVL          40    70.0   
3                                             BAMS           0     NaN   
4  BDS, MDS - Conservative Dentistry & Endodontics          16   100.0   

              Place           Profile  \
0    Ghatkopar East  General Medicine   
1  West Marredpally    ENT Specialist   
2          KK Nagar    Dermatologists   
3   New Ashok Nagar          Ayurveda   
4    Kanakpura Road           Dentist   

                                  Miscellaneous_Info        City  
0                                                NaN      Mumbai  
1                                                NaN   Hyderabad  
2                   70% 4 Feedback KK Nagar, Chennai     Chennai  
3                                                NaN       Delhi  
4  General Dentistry Conservative Dentistry Cosme...   Bangalore

We can see above that "City" column has been added and values of "Place" column is splitted into both these two columns.

Now, trying treating Miscellaneous_Info:
    I find only number of feedback in this column that could be little relevant for Model Building.

In [55]:
# trying to seperate number of feedbacks from Miscellaneous_Info:

data['Miscellaneous_Info'] = data.Miscellaneous_Info.str.extract('% (.+) Feedback')

In [56]:
data.head()

Qualification  Experience  Rating              Place  \
0           BHMS, MD - Homeopathy          24   100.0           Kakkanad   
1    BAMS, MD - Ayurveda Medicine          12    98.0         Whitefield   
2  MBBS, MS - Otorhinolaryngology           9     NaN    Mathikere - BEL   
3             BSc - Zoology, BAMS          12     NaN  Bannerghatta Road   
4                            BAMS          20   100.0       Keelkattalai   

          Profile Miscellaneous_Info  Fees       City  
0       Homeopath                 16   100  Ernakulam  
1        Ayurveda                 76   350  Bangalore  
2  ENT Specialist                NaN   300  Bangalore  
3        Ayurveda                NaN   250  Bangalore  
4        Ayurveda                  4   250    Chennai

In [57]:
data_test['Miscellaneous_Info'] = data_test.Miscellaneous_Info.str.extract('% (.+) Feedback')

In [58]:
data_test.head()

Qualification  Experience  Rating  \
0                                             MBBS          35     NaN   
1       MBBS, Diploma in Otorhinolaryngology (DLO)          31     NaN   
2                                       MBBS, DDVL          40    70.0   
3                                             BAMS           0     NaN   
4  BDS, MDS - Conservative Dentistry & Endodontics          16   100.0   

              Place           Profile Miscellaneous_Info        City  
0    Ghatkopar East  General Medicine                NaN      Mumbai  
1  West Marredpally    ENT Specialist                NaN   Hyderabad  
2          KK Nagar    Dermatologists                  4     Chennai  
3   New Ashok Nagar          Ayurveda                NaN       Delhi  
4    Kanakpura Road           Dentist                NaN   Bangalore

In [59]:
# checking NULL values:

data.isnull().sum()

Qualification            0
Experience               0
Rating                3302
Place                   25
Profile                  0
Miscellaneous_Info    4461
Fees                     0
City                    26
dtype: int64

In [60]:
data_test.isnull().sum()

Qualification            0
Experience               0
Rating                1090
Place                    6
Profile                  0
Miscellaneous_Info    1436
City                     6
dtype: int64

In [61]:
data_test.shape

(1987, 7)

Now, if we see that earlier Miscellaneous_Info had 2620 NaN values in training dataset , which now has been increased to 4461 almost 75% of the size of the data.
So, it would not provide us any significant information. Hence better to leave it.

In [62]:
# Dropping No. of Feedback from the dataset:

data.drop("Miscellaneous_Info", axis = 1, inplace = True)

data_test.drop("Miscellaneous_Info", axis = 1, inplace = True)

In [63]:
# Renaming the columns:

data = data.rename(columns = {'Experience': 'Experience(in years)','Rating': 'Rating(in percent)' }, inplace = False)
data_test = data_test.rename(columns = {'Experience': 'Experience(in years)','Rating': 'Rating(in percent)'}, inplace = False)

In [64]:
# just to check the data :
data.head()

Qualification  Experience(in years)  Rating(in percent)  \
0           BHMS, MD - Homeopathy                    24               100.0   
1    BAMS, MD - Ayurveda Medicine                    12                98.0   
2  MBBS, MS - Otorhinolaryngology                     9                 NaN   
3             BSc - Zoology, BAMS                    12                 NaN   
4                            BAMS                    20               100.0   

               Place         Profile  Fees       City  
0           Kakkanad       Homeopath   100  Ernakulam  
1         Whitefield        Ayurveda   350  Bangalore  
2    Mathikere - BEL  ENT Specialist   300  Bangalore  
3  Bannerghatta Road        Ayurveda   250  Bangalore  
4       Keelkattalai        Ayurveda   250    Chennai

In [65]:
data_test.head()

Qualification  Experience(in years)  \
0                                             MBBS                    35   
1       MBBS, Diploma in Otorhinolaryngology (DLO)                    31   
2                                       MBBS, DDVL                    40   
3                                             BAMS                     0   
4  BDS, MDS - Conservative Dentistry & Endodontics                    16   

   Rating(in percent)             Place           Profile        City  
0                 NaN    Ghatkopar East  General Medicine      Mumbai  
1                 NaN  West Marredpally    ENT Specialist   Hyderabad  
2                70.0          KK Nagar    Dermatologists     Chennai  
3                 NaN   New Ashok Nagar          Ayurveda       Delhi  
4               100.0    Kanakpura Road           Dentist   Bangalore

In [66]:
# Grouping/ Sorting data by Profile:

data = data.sort_values(by="Profile",ascending=True)
data_test = data_test.sort_values(by="Profile",ascending=True)

In [67]:
data_test.head()

Qualification  Experience(in years)  \
993                                                BAMS                     8   
1660  BAMS, MD - Internal Medicine, Diploma in Emerg...                    28   
665                              MD - Ayurveda Medicine                    13   
259                                                BAMS                    42   
668                                                BAMS                    31   

      Rating(in percent)           Place   Profile        City  
993                  NaN      Chattarpur  Ayurveda       Delhi  
1660                95.0    Kalyan Nagar  Ayurveda   Bangalore  
665                  NaN          Rohini  Ayurveda       Delhi  
259                  NaN  Vileparle East  Ayurveda      Mumbai  
668                  NaN       Yelahanka  Ayurveda   Bangalore

In [68]:
data.head()

Qualification  Experience(in years)  Rating(in percent)  \
1757  MD - Ayurveda Medicine, BAMS                     9                 NaN   
563   BAMS, MD - Ayurveda Medicine                    43                 NaN   
4648                          BAMS                    10                 NaN   
3688                          BAMS                    10                90.0   
1449                          BAMS                     9                 NaN   

               Place   Profile  Fees       City  
1757         Balapur  Ayurveda   200  Hyderabad  
563     Andheri East  Ayurveda   100     Mumbai  
4648   Jubilee Hills  Ayurveda   100  Hyderabad  
3688        Borivali  Ayurveda   100     Mumbai  
1449  Rajouri Garden  Ayurveda   500      Delhi

In [70]:
# Extracting relevant qualification

data["Qualification"] = data["Qualification"].str.split("," ,expand = True,)

Qualification = {}
for x in data["Qualification"].values:
    for each in x:
        each = each.strip()
        if each in Qualification:
            
            Qualification[each]=+1
        else:
                Qualification[each] = 1

In [72]:
data.head(15)

Qualification  Experience(in years)  Rating(in percent)  \
1757  MD - Ayurveda Medicine                     9                 NaN   
563                     BAMS                    43                 NaN   
4648                    BAMS                    10                 NaN   
3688                    BAMS                    10                90.0   
1449                    BAMS                     9                 NaN   
567                     BAMS                    11                 NaN   
4635                    BAMS                     2                 NaN   
4629                    BAMS                     8                 NaN   
2511                    BAMS                     9                88.0   
3012                    BAMS                     8                 NaN   
3697                    BAMS                    11                 NaN   
3007                    BAMS                    11                 NaN   
4620                    BAMS                     9                98.0   
4619                    BAMS                    29               100.0   
3699                    BAMS                    27                 NaN   

                 Place   Profile  Fees       City  
1757           Balapur  Ayurveda   200  Hyderabad  
563       Andheri East  Ayurveda   100     Mumbai  
4648     Jubilee Hills  Ayurveda   100  Hyderabad  
3688          Borivali  Ayurveda   100     Mumbai  
1449    Rajouri Garden  Ayurveda   500      Delhi  
567   Mayur Vihar Ph-I  Ayurveda   500      Delhi  
4635           T Nagar  Ayurveda   300    Chennai  
4629    Ghatkopar East  Ayurveda   200     Mumbai  
2511    Mira Bhayandar  Ayurveda   150     Mumbai  
3012       Malleswaram  Ayurveda   250  Bangalore  
3697      Thoraipakkam  Ayurveda   200    Chennai  
3007    Mumbai Central  Ayurveda   100     Mumbai  
4620     Borivali West  Ayurveda   300     Mumbai  
4619    Vileparle East  Ayurveda   500     Mumbai  
3699    Vileparle East  Ayurveda   300     Mumbai

In [74]:
data['Qualification'].unique()

array(['MD - Ayurveda Medicine', 'BAMS',
       'MD - Social & Preventive Medicine / Community Medicine',
       'MD - General Medicine', 'B.Sc', 'MS - Ayurveda', 'BSAM',
       'MD - Anatomy', 'PhD - Neuropsychology', 'Ayurvedic panchkarma',
       'BUMS', 'MD - Medicine',
       'MD - Ayurvedic Obstetrics and Gynecology', 'G.A.M.S',
       'Certified In Panchkarma', 'DNHE', 'CGO', 'PGD ND',
       'MD - Ayurvedic Pharmacology', 'BIMS', 'BSc - Zoology',
       'Diploma in Trichology - Cosmetology', 'MD - Acupuncture',
       'MS - Ayurvedic Gynecology & Obstetrics', 'MD AYURVED AND YOG',
       'MBBS', 'BDS', 'MDS - Periodontology and Oral Implantology',
       'MDS - Orthodontics', 'MDS - Oral & Maxillofacial Surgery',
       'MDS - Pedodontics', 'MDS - Prosthodontics',
       'MDS - Prosthodontist And Crown Bridge',
       'MDS - Paedodontics And Preventive Dentistry',
       'MDS - Periodontics', 'MDS - Conservative Dentistry & Endodontics',
       'MDS - Oral & Maxillofacial Patho

In [75]:
data['Qualification'].nunique()

151

Still the unique values in the qualification are 151.

In [76]:
# Extracting relevant qualification

data_test["Qualification"] = data_test["Qualification"].str.split("," ,expand = True,)

Qualification = {}
for x in data_test["Qualification"].values:
    for each in x:
        each = each.strip()
        if each in Qualification:
            
            Qualification[each]=+1
        else:
                Qualification[each] = 1

In [78]:
# Rearranging the columns :

data = data[['Profile', 'Qualification', 'Experience(in years)', 'Rating(in percent)', 'Place','City','Fees']]

data_test = data_test[['Profile', 'Qualification', 'Experience(in years)', 'Rating(in percent)', 'Place','City']]

In [79]:
data.head()

Profile           Qualification  Experience(in years)  \
1757  Ayurveda  MD - Ayurveda Medicine                     9   
563   Ayurveda                    BAMS                    43   
4648  Ayurveda                    BAMS                    10   
3688  Ayurveda                    BAMS                    10   
1449  Ayurveda                    BAMS                     9   

      Rating(in percent)           Place       City  Fees  
1757                 NaN         Balapur  Hyderabad   200  
563                  NaN    Andheri East     Mumbai   100  
4648                 NaN   Jubilee Hills  Hyderabad   100  
3688                90.0        Borivali     Mumbai   100  
1449                 NaN  Rajouri Garden      Delhi   500

In [80]:
data_test.head()

Profile           Qualification  Experience(in years)  \
993   Ayurveda                    BAMS                     8   
1660  Ayurveda                    BAMS                    28   
665   Ayurveda  MD - Ayurveda Medicine                    13   
259   Ayurveda                    BAMS                    42   
668   Ayurveda                    BAMS                    31   

      Rating(in percent)           Place        City  
993                  NaN      Chattarpur       Delhi  
1660                95.0    Kalyan Nagar   Bangalore  
665                  NaN          Rohini       Delhi  
259                  NaN  Vileparle East      Mumbai  
668                  NaN       Yelahanka   Bangalore

In [81]:
data.dtypes

Profile                  object
Qualification            object
Experience(in years)      int32
Rating(in percent)      float64
Place                    object
City                     object
Fees                      int64
dtype: object

In [82]:
# Identifying the top 20 most occured qualification and creating dummy variables for those 20 qualifications:

most_occ = sorted(data['Qualification'].items(), key = lambda x:x[1],reverse=True)[:20]

final_qual = []

for tup in most_occ:
    final_qual.append(tup[0])
    
for title in final_qual:
    data[title]=0
    
for x,y in zip(data["Qualification"].values, np.array([idx for idx in range(len(data))])):
    for q in x:
        q = q.strip()
        if q in final_qual:
            data[q][y] = 1

In [83]:
# Identifying the top 20 most occured qualification and creating dummy variables for those 20 qualifications:

most_occ = sorted(data_test['Qualification'].items(), key = lambda x:x[1],reverse=True)[:20]

final_qual = []

for tup in most_occ:
    final_qual.append(tup[0])
    
for title in final_qual:
    data_test[title]=0
    
for x,y in zip(data_test["Qualification"].values, np.array([idx for idx in range(len(data_test))])):
    for q in x:
        q = q.strip()
        if q in final_qual:
            data_test[q][y] = 1

In [84]:
# checking the data:
data.head(15)

Profile           Qualification  Experience(in years)  \
1757  Ayurveda  MD - Ayurveda Medicine                     9   
563   Ayurveda                    BAMS                    43   
4648  Ayurveda                    BAMS                    10   
3688  Ayurveda                    BAMS                    10   
1449  Ayurveda                    BAMS                     9   
567   Ayurveda                    BAMS                    11   
4635  Ayurveda                    BAMS                     2   
4629  Ayurveda                    BAMS                     8   
2511  Ayurveda                    BAMS                     9   
3012  Ayurveda                    BAMS                     8   
3697  Ayurveda                    BAMS                    11   
3007  Ayurveda                    BAMS                    11   
4620  Ayurveda                    BAMS                     9   
4619  Ayurveda                    BAMS                    29   
3699  Ayurveda                    BAMS                    27   

      Rating(in percent)             Place       City  Fees  1715  2505  2420  \
1757                 NaN           Balapur  Hyderabad   200     0     0     0   
563                  NaN      Andheri East     Mumbai   100     0     0     0   
4648                 NaN     Jubilee Hills  Hyderabad   100     0     0     0   
3688                90.0          Borivali     Mumbai   100     0     0     0   
1449                 NaN    Rajouri Garden      Delhi   500     0     0     0   
567                  NaN  Mayur Vihar Ph-I      Delhi   500     0     0     0   
4635                 NaN           T Nagar    Chennai   300     0     0     0   
4629                 NaN    Ghatkopar East     Mumbai   200     0     0     0   
2511                88.0    Mira Bhayandar     Mumbai   150     0     0     0   
3012                 NaN       Malleswaram  Bangalore   250     0     0     0   
3697                 NaN      Thoraipakkam    Chennai   200     0     0     0   
3007                 NaN    Mumbai Central     Mumbai   100     0     0     0   
4620                98.0     Borivali West     Mumbai   300     0     0     0   
4619               100.0    Vileparle East     Mumbai   500     0     0     0   
3699                 NaN    Vileparle East     Mumbai   300     0     0     0   

      ...  1402  401  174  1221  104  514  2030  4303  2028  5631  
1757  ...     0    0    0     0    0    0     0     0     0     0  
563   ...     0    0    0     0    0    0     0     0     0     0  
4648  ...     0    0    0     0    0    0     0     0     0     0  
3688  ...     0    0    0     0    0    0     0     0     0     0  
1449  ...     0    0    0     0    0    0     0     0     0     0  
567   ...     0    0    0     0    0    0     0     0     0     0  
4635  ...     0    0    0     0    0    0     0     0     0     0  
4629  ...     0    0    0     0    0    0     0     0     0     0  
2511  ...     0    0    0     0    0    0     0     0     0     0  
3012  ...     0    0    0     0    0    0     0     0     0     0  
3697  ...     0    0    0     0    0    0     0     0     0     0  
3007  ...     0    0    0     0    0    0     0     0     0     0  
4620  ...     0    0    0     0    0    0     0     0     0     0  
4619  ...     0    0    0     0    0    0     0     0     0     0  
3699  ...     0    0    0     0    0    0     0     0     0     0  

[15 rows x 27 columns]

In [85]:
data.drop("Qualification", axis = 1, inplace = True)
data_test.drop("Qualification", axis = 1, inplace = True)

In [86]:
data_test.head(15)

Profile  Experience(in years)  Rating(in percent)           Place  \
993   Ayurveda                     8                 NaN      Chattarpur   
1660  Ayurveda                    28                95.0    Kalyan Nagar   
665   Ayurveda                    13                 NaN          Rohini   
259   Ayurveda                    42                 NaN  Vileparle East   
668   Ayurveda                    31                 NaN       Yelahanka   
1673  Ayurveda                    17                 NaN   Paschim Vihar   
1137  Ayurveda                     8                90.0        Madhapur   
1675  Ayurveda                    19                 NaN       Velachery   
263   Ayurveda                    29                 NaN  Vileparle West   
252   Ayurveda                    28                 NaN           Delhi   
249   Ayurveda                    11                 NaN     Madambakkam   
247   Ayurveda                    14                 NaN     Malleswaram   
1682  Ayurveda                    25                 NaN       Khar East   
674   Ayurveda                    23                 NaN     Mulund West   
1134  Ayurveda                     8                 NaN        Mylapore   

            City  89  991  1571  1583  570  ...  867  1465  1881  1110  1152  \
993        Delhi   0    0     0     0    0  ...    0     0     0     0     0   
1660   Bangalore   0    0     0     0    0  ...    0     0     0     0     0   
665        Delhi   0    0     0     0    0  ...    0     0     0     0     0   
259       Mumbai   0    0     0     0    0  ...    0     0     0     0     0   
668    Bangalore   0    0     0     0    0  ...    0     0     0     0     0   
1673       Delhi   0    0     0     0    0  ...    0     0     0     0     0   
1137   Hyderabad   0    0     0     0    0  ...    0     0     0     0     0   
1675     Chennai   0    0     0     0    0  ...    0     0     0     0     0   
263       Mumbai   0    0     0     0    0  ...    0     0     0     0     0   
252        Delhi   0    0     0     0    0  ...    0     0     0     0     0   
249      Chennai   0    0     0     0    0  ...    0     0     0     0     0   
247    Bangalore   0    0     0     0    0  ...    0     0     0     0     0   
1682      Mumbai   0    0     0     0    0  ...    0     0     0     0     0   
674       Mumbai   0    0     0     0    0  ...    0     0     0     0     0   
1134     Chennai   0    0     0     0    0  ...    0     0     0     0     0   

      1878  1922  181  597  1765  
993      0     0    0    0     0  
1660     0     0    0    0     0  
665      0     0    0    0     0  
259      0     0    0    0     0  
668      0     0    0    0     0  
1673     0     0    0    0     0  
1137     0     0    0    0     0  
1675     0     0    0    0     0  
263      0     0    0    0     0  
252      0     0    0    0     0  
249      0     0    0    0     0  
247      0     0    0    0     0  
1682     0     0    0    0     0  
674      0     0    0    0     0  
1134     0     0    0    0     0  

[15 rows x 25 columns]

Now, Dealing with Null Values in both the data sets:
    

In [87]:
data.isnull().sum()

Profile                    0
Experience(in years)       0
Rating(in percent)      3302
Place                     25
City                      26
Fees                       0
1715                       0
2505                       0
2420                       0
236                        0
2084                       0
3937                       0
623                        0
1071                       0
5649                       0
3897                       0
1402                       0
401                        0
174                        0
1221                       0
104                        0
514                        0
2030                       0
4303                       0
2028                       0
5631                       0
dtype: int64

In [88]:
data_test.isnull().sum()

Profile                    0
Experience(in years)       0
Rating(in percent)      1090
Place                      6
City                       6
89                         0
991                        0
1571                       0
1583                       0
570                        0
1814                       0
526                        0
1656                       0
931                        0
1515                       0
867                        0
1465                       0
1881                       0
1110                       0
1152                       0
1878                       0
1922                       0
181                        0
597                        0
1765                       0
dtype: int64

In [89]:
data['Place'].fillna('missing', inplace=True)
data_test['Place'].fillna('missing', inplace=True)

In [90]:
data['Rating(in percent)'].fillna('0', inplace=True)
data_test['Rating(in percent)'].fillna('0', inplace=True)

In [91]:
data['City'].fillna('missing', inplace=True)
data_test['City'].fillna('missing', inplace=True)

In [92]:
data.isnull().sum()

Profile                 0
Experience(in years)    0
Rating(in percent)      0
Place                   0
City                    0
Fees                    0
1715                    0
2505                    0
2420                    0
236                     0
2084                    0
3937                    0
623                     0
1071                    0
5649                    0
3897                    0
1402                    0
401                     0
174                     0
1221                    0
104                     0
514                     0
2030                    0
4303                    0
2028                    0
5631                    0
dtype: int64

In [93]:
data.head()

Profile  Experience(in years) Rating(in percent)           Place  \
1757  Ayurveda                     9                  0         Balapur   
563   Ayurveda                    43                  0    Andheri East   
4648  Ayurveda                    10                  0   Jubilee Hills   
3688  Ayurveda                    10                 90        Borivali   
1449  Ayurveda                     9                  0  Rajouri Garden   

           City  Fees  1715  2505  2420  236  ...  1402  401  174  1221  104  \
1757  Hyderabad   200     0     0     0    0  ...     0    0    0     0    0   
563      Mumbai   100     0     0     0    0  ...     0    0    0     0    0   
4648  Hyderabad   100     0     0     0    0  ...     0    0    0     0    0   
3688     Mumbai   100     0     0     0    0  ...     0    0    0     0    0   
1449      Delhi   500     0     0     0    0  ...     0    0    0     0    0   

      514  2030  4303  2028  5631  
1757    0     0     0     0     0  
563     0     0     0     0     0  
4648    0     0     0     0     0  
3688    0     0     0     0     0  
1449    0     0     0     0     0  

[5 rows x 26 columns]

In [94]:
data = pd.get_dummies(data, columns=['Profile'])
data_test  = pd.get_dummies(data_test,  columns=['Profile'])

In [95]:
data.head()

Experience(in years) Rating(in percent)           Place       City  \
1757                     9                  0         Balapur  Hyderabad   
563                     43                  0    Andheri East     Mumbai   
4648                    10                  0   Jubilee Hills  Hyderabad   
3688                    10                 90        Borivali     Mumbai   
1449                     9                  0  Rajouri Garden      Delhi   

      Fees  1715  2505  2420  236  2084  ...  2030  4303  2028  5631  \
1757   200     0     0     0    0     0  ...     0     0     0     0   
563    100     0     0     0    0     0  ...     0     0     0     0   
4648   100     0     0     0    0     0  ...     0     0     0     0   
3688   100     0     0     0    0     0  ...     0     0     0     0   
1449   500     0     0     0    0     0  ...     0     0     0     0   

      Profile_Ayurveda  Profile_Dentist  Profile_Dermatologists  \
1757                 1                0                       0   
563                  1                0                       0   
4648                 1                0                       0   
3688                 1                0                       0   
1449                 1                0                       0   

      Profile_ENT Specialist  Profile_General Medicine  Profile_Homeopath  
1757                       0                         0                  0  
563                        0                         0                  0  
4648                       0                         0                  0  
3688                       0                         0                  0  
1449                       0                         0                  0  

[5 rows x 31 columns]